samples/application_development/out_of_tree_driver/src/hello_world_driver.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __HELLO_WORLD_DRIVER_H__
+#define __HELLO_WORLD_DRIVER_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <device.h>
+
+/*
+ * This 'Hello World' driver has a 'print' syscall that prints the
```

Comment:
I think adding this sample is a great idea, thanks for working on it.

Would it make sense to make the driver source code a separate module, to really make sure there's no special magic due to being inside the zephyr tree?

Reply:
You mean using ZEPHYR_EXTRA_MODULES to point to a dir inside of the app dir, right?

Not an entry in west.yml?

Reply:
I was asking about an entry in west.yml, yes.

Reply:
Oh, I would not like to do that as I don't want to slow down normal west usage just for a sample.

Reply:
OK, fair enough.

Would ZEPHYR_EXTRA_MODULES work for code that's nested inside the zephyr repository? It would be nice to make the sample work as closely as possible to how users will need to use it.

Reply:
Yes

Reply:
Ok, I think that would be nice, then.  But I don't think it should block the merge. (A comment describing how ZEPHYR_EXTRA_MODULES would not be needed if the user actually wants to make a truly out of tree driver would also be good in that case.)

Reply:
> (A comment describing how ZEPHYR_EXTRA_MODULES would not be needed if the user actually wants to make a truly out of tree driver would also be good in that case.)

By the way, I saw a user on slack confused about duplicate items in their ZEPHYR_MODULES list (caused by adding some directories that were already in ZEPHYR_MODULES to their custom ZEPHYR_EXTRA_MODULES) led to errors.

Would it make sense for CMake to deduplicate this list? That would make it possible for users to copy/paste the sample code and ignore that comment.

Reply:
It does make sense to deduplicate this list. A PR to this effect would be welcome.

Reply:
Organized the driver as a ZEPHYR_EXTRA_MODULE.